### DISCUSSION TITLE
Move import to module scope

### DISCUSSION THREAD

**Discussion by dmartin** [microsoft/pylance-release#5789] [Enhancement]
Sometimes, in the middle of writing a method or function, it turns out that you need to reference an item which has not been imported yet. When auto-import or "add import" code actions work, it's no problem. But those systems don't work in all cases, and when do you need to manually add a top-level import, it requires a bit of a mental context switch.

 Here's a (contrived) example:

```python
from django.db.models import Model

class Company(Model):
    def employees(self, include_inactive: bool = False):
        queryset = self.employee_set.all()
        if not include_inactive:
            queryset = queryset.exclude(Q(active=False) | Q(on_leave=True))
        return queryset
```

Say you just typed the second-to-last line. `django.db.models.Q` has not been imported yet, so there will be errors on that line. There is no Quick Fix action available to import `Q` (in this case, I believe it has to do with Django stub issues, but the point is any general case where a matching import is not detected).

In this short example, it's obviously no problem to add an import for `Q`. But what if there are 700 lines above `CompanyQuerySet`? Then the options are:

- Scroll to the top of the file to manually add the import, losing current position
- Split the editor and scroll to top in the duplicate (fine, but clunky)
- Write it as an inline method-scoped import

It would be nice if it were possible to write out the manual import inline, then apply a refactoring action to move the import to module scope (i.e. "Move import to top of file").

**Comment by rchiodo** [comment]
Thanks for the suggestion. Moving to discussions for up votes.
